odoo/service/server.py

```diff
@@ -888,6 +888,8 @@ def stop(self, graceful=True):
             limit = time.time() + self.timeout
             for pid in self.workers:
                 self.worker_kill(pid, signal.SIGINT)
+            if self.socket:
+                self.socket.close()
```

Comment:
Shouldn't the exact same code at lines 905-906 be removed?

Reply:
Not necessarily, it depends if we want to be able to call `stop(graceful=False)` without having called `stop(graceful=True)` first.

(Calling `socket.close()` on a closed socket doesn't do anything anyway)

Reply:
Ah yes I'd not noticed the difference of indentation.

But then wouldn't it work out to stop the socket before signaling the workers anyway? aka close between lines 884 and 885?

Reply:
+1 that stop(graceful=False) should close the server socket, lgtm

Reply:
> But then wouldn't it work out to stop the socket before signaling the workers anyway? aka close between lines 884 and 885?

Yes perhaps, maybe we wouldn't even need to signal them to shut down either if they do so when noticing the closed socket :stuck_out_tongue:

Reply:
> Yes perhaps, maybe we wouldn't even need to signal them to shut down either if they do so when noticing the closed socket stuck_out_tongue

Is there any way to do that? As far as the workers are concerned the socket is still open and that's why we need to shoot them no?

Reply:
> Is there any way to do that? As far as the workers are concerned the socket is still open and that's why we need to shoot them no?

Myeah, there'd be an easy way but didn't find one :(

I'll put the socket.close() before the `if graceful:` condition as you suggested.